lib/rgeo/geos/capi_factory.rb

```diff
@@ -235,10 +235,10 @@ def encode_with(coder_)  # :nodoc:
         coder_['srid'] = _srid
         coder_['buffer_resolution'] = _buffer_resolution
         coder_['lenient_multi_polygon_assertions'] = (_flags & 0x1 != 0)
-        coder_['wkt_generator'] = _wkt_generator._properties
-        coder_['wkb_generator'] = _wkb_generator._properties
-        coder_['wkt_parser'] = _wkt_parser._properties
-        coder_['wkb_parser'] = _wkb_parser._properties
+        coder_['wkt_generator'] = _wkt_generator ? _wkt_generator._properties : {}
+        coder_['wkb_generator'] = _wkb_generator ? _wkb_generator._properties : {}
+        coder_['wkt_parser'] = _wkt_parser ? _wkt_parser._properties : {}
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.